metis-enrichment/metis-enrichment-service/src/main/java/eu/europeana/enrichment/service/EntityService.java

```diff
@@ -23,18 +25,26 @@ public EntityService(String mongoHost, int mongoPort) {
 		this.mongoPort = mongoPort;
 	}
 
-	public OrganizationTermList storeOrganization(Organization org) {
+	public OrganizationTermList storeOrganization(Organization org,
+			Date created, Date modified) {
 		MongoDatabaseUtils.dbExists(mongoHost, mongoPort);
 
-		// delete old references
-		MongoDatabaseUtils.deleteOrganizations(org.getAbout());
-
 		// build term list
 		OrganizationTermList termList = organizationToOrganizationTermList(
-				(OrganizationImpl) org);
-		Date now = new Date();
-		termList.setCreated(now);
-		termList.setModified(now);
+				(OrganizationImpl) org, created, modified);
+
+		MongoTermList<ContextualClassImpl> storedOrg = MongoDatabaseUtils
+				.findByCode(org.getAbout(), EntityClass.ORGANIZATION);
+
+		// it is an update
+		if (storedOrg != null) {
+			// set database id
+			termList.setId(storedOrg.getId());
```

Comment:
This variable can actually be null at this point. Could we check nullity somewhere?

Reply:
It is not needed. This is the only operation that needs to be made in the case that the organization already exists in the database. If not, mongojack generates a new id. I hoped that the two line of comments are enough to explain why.